rest_framework_json_api/schemas/openapi.py

```diff
@@ -0,0 +1,839 @@
+        """
+        return [{'$ref': '#/components/parameters/include'}]
+
+    def _get_fields_parameters(self, path, method):
+        """
+        sparse fieldsets https://jsonapi.org/format/#fetching-sparse-fieldsets
+        """
+        return [{'$ref': '#/components/parameters/fields'}]
+
+    def _get_sort_parameters(self, path, method):
+        """
+        sort parameter: https://jsonapi.org/format/#fetching-sorting
+        """
+        return [{'$ref': '#/components/parameters/sort'}]
+
+    def _get_collection(self, operation):
+        """
+        jsonapi-structured 200 response for GET of a collection
+        """
+        operation['responses'] = {
+            '200': self._get_toplevel_200_response(operation, collection=True)
+        }
+        self._add_get_4xx_responses(operation)
+
+    def _get_item(self, operation):
+        """ jsonapi-structured response for GET of an item """
+        operation['responses'] = {
+            '200': self._get_toplevel_200_response(operation, collection=False)
+        }
+        self._add_get_4xx_responses(operation)
+
+    def _get_toplevel_200_response(self, operation, collection=True):
+        """ top-level JSONAPI GET 200 response """
+        if collection:
+            data = {'type': 'array', 'items': self._get_item_schema(operation)}
+        else:
+            data = self._get_item_schema(operation)
+
+        return {
+            'description': operation['operationId'],
+            'content': {
+                'application/vnd.api+json': {
+                    'schema': {
+                        'type': 'object',
+                        'required': ['data'],
+                        'properties': {
+                            'data': data,
+                            'included': {
+                                'type': 'array',
+                                'uniqueItems': True,
+                                'items': {
+                                    '$ref': '#/components/schemas/resource'
+                                }
+                            },
+                            'links': {
+                                'description': 'Link members related to primary data',
+                                'allOf': [
+                                    {'$ref': '#/components/schemas/links'},
+                                    {'$ref': '#/components/schemas/pagination'}
+                                ]
+                            },
+                            'jsonapi': {
+                                '$ref': '#/components/schemas/jsonapi'
+                            }
+                        }
+                    }
+                }
+            }
+        }
+
+    def _get_item_schema(self, operation):
+        """
+        get the schema for item
+        """
+        content = {}
+        view = self.view
+        if hasattr(view, 'get_serializer'):
+            try:
+                serializer = view.get_serializer()
```

Comment:
upstream there is a method `_get_serializer` which does this error handling.

Reply:
I've fixed this to use upstream's `get_serializer`